lib/cocoapods/installer/xcode/pods_project_generator.rb

```diff
@@ -258,6 +259,23 @@ def configure_app_extension_api_only_for_target(target)
             config.build_settings['APPLICATION_EXTENSION_API_ONLY'] = 'YES'
           end
         end
+
+        def user_swift_version
+          declared_swift_versions = analysis_result
+            .targets
+            .map(&:user_project)
+            .compact
+            .flat_map(&:native_targets)
+            .flat_map(&:build_configurations)
+            .flat_map(&:build_settings)
+            .map { |settings| settings['SWIFT_VERSION'] }
+            .compact
+            .uniq
+
+          # TODO: Assert only 1 or 0 unique Swift Versions declared
```

Comment:
I'm gonna look into this tonight/tomorrow. Been a busy week and I'm not 100% sure on how best to do it.